plugins/inputs/ravendb/ravendb.go

```diff
@@ -0,0 +1,680 @@
+package ravendb
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/influxdata/telegraf/plugins/common/tls"
+	"net/http"
+	"net/url"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/internal"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+// defaultURL will set a default value that corresponds to the default value
+// used by RavenDB
+const DefaultURL = "http://localhost:8080"
+
+// Default http timeouts
+const DefaultResponseHeaderTimeout = 3
+const DefaultClientTimeout = 4
+
+type gatherFunc func(r *RavenDB, acc telegraf.Accumulator)
+
+var gatherFunctions = []gatherFunc{gatherServer, gatherDatabases, gatherIndexes, gatherCollections}
+
+// RavenDB defines the configuration necessary for gathering metrics,
+// see the sample config for further details
+type RavenDB struct {
+	URL  string `toml:"url"`
+	Name string `toml:"name"`
+	tls.ClientConfig
+
+	Log telegraf.Logger
+
+	ResponseHeaderTimeout internal.Duration `toml:"header_timeout"`
+	ClientTimeout         internal.Duration `toml:"client_timeout"`
+
+	GatherServerStats     bool     `toml:"gather_server_stats"`
+	GatherDbStats         bool     `toml:"gather_db_stats"`
+	GatherIndexStats      bool     `toml:"gather_index_stats"`
+	GatherCollectionStats bool     `toml:"gather_collection_stats"`
+	DbStatsDbs            []string `toml:"db_stats_dbs"`
+	IndexStatsDbs         []string `toml:"index_stats_dbs"`
+	CollectionStatsDbs    []string `toml:"collection_stats_dbs"`
```

Comment:
Please group the user-definable options. Move `Log telegraf.Logger` at the end of the user-definable options and add `toml:"-"` to prevent toml to try to set this.

Reply:
I'm not sure I fully understand. I end up with this:
```
type RavenDB struct {
	URL  string `toml:"url"`
	Name string `toml:"name"`
	tls.ClientConfig

	ResponseHeaderTimeout internal.Duration `toml:"header_timeout"`
	ClientTimeout         internal.Duration `toml:"client_timeout"`

	GatherServerStats     bool     `toml:"gather_server_stats"`
	GatherDbStats         bool     `toml:"gather_db_stats"`
	GatherIndexStats      bool     `toml:"gather_index_stats"`
	GatherCollectionStats bool     `toml:"gather_collection_stats"`
	DbStatsDbs            []string `toml:"db_stats_dbs"`
	IndexStatsDbs         []string `toml:"index_stats_dbs"`
	CollectionStatsDbs    []string `toml:"collection_stats_dbs"`

	Client *http.Client `toml:"-"`
	Log telegraf.Logger `toml:"-"`
}
```

Reply:
Well almost. Usually the user-configurable options (everything with a `toml` annotation) are listed first, `Log` is the last in this group. Afterwards,  `tls.ClientConfig` and internal fields (e.g. `client`) follow. That's the usual way so it's easier for developers they feel home in this plugin. :-)

Reply:
ok updated

Reply:
Please mark as resolved.